src/m365/spo/commands/homesite/homesite-set.ts

```diff
@@ -77,26 +70,151 @@ class SpoHomeSiteSetCommand extends SpoCommand {
-        requestOptions.data.configuration = { vivaConnectionsDefaultStart: args.options.vivaConnectionsDefaultStart };
+      const homeSiteCount = await this.getHomeSiteCount(spoAdminUrl, logger);
+
+      if (args.options.vivaConnectionsDefaultStart !== undefined ||
+        args.options.draftMode !== undefined ||
+        args.options.audienceIds !== undefined ||
+        args.options.audienceNames !== undefined ||
+        args.options.targetedLicenseType !== undefined ||
+        args.options.order !== undefined) {
+
+        const configuration: any = {};
+        if (args.options.vivaConnectionsDefaultStart !== undefined) {
+          configuration.IsVivaConnectionsDefaultStartPresent = true;
+          configuration.vivaConnectionsDefaultStart = args.options.vivaConnectionsDefaultStart;
+        }
+        if (args.options.draftMode !== undefined) {
+          configuration.IsInDraftModePresent = true;
+          configuration.isInDraftMode = args.options.draftMode;
+        }
+        if (args.options.audienceIds !== undefined) {
+          configuration.IsAudiencesPresent = true;
+          configuration.Audiences = args.options.audienceIds.split(',').map(id => id.trim());
+        }
+        if (args.options.audienceNames !== undefined) {
+          configuration.IsAudiencesPresent = true;
+          configuration.Audiences = await this.transformAudienceNamesToIds(args.options.audienceNames);
+        }
+        if (args.options.targetedLicenseType !== undefined) {
+          configuration.IsTargetedLicenseTypePresent = true;
+          configuration.TargetedLicenseType = this.convertTargetedLicenseTypeToNumber(args.options.targetedLicenseType);
+        }
+        if (args.options.order !== undefined) {
+          configuration.IsOrderPresent = true;
+          configuration.Order = args.options.order;
+        }
+        const hasOnlyVivaConnectionsDefaultStart = args.options.vivaConnectionsDefaultStart !== undefined &&
+          args.options.draftMode === undefined &&
+          args.options.audienceIds === undefined &&
+          args.options.audienceNames === undefined &&
+          args.options.targetedLicenseType === undefined &&
+          args.options.order === undefined;
+        if (homeSiteCount <= 1 && hasOnlyVivaConnectionsDefaultStart) {
+          requestOptions.url += '/SetSPHSiteWithConfiguration';
+          requestOptions.data.sphSiteUrl = args.options.siteUrl;
+          requestOptions.data.configuration = { vivaConnectionsDefaultStart: args.options.vivaConnectionsDefaultStart };;
+        }
+        else {
+          requestOptions.url += '/UpdateTargetedSite';
+          requestOptions.data.siteUrl = args.options.siteUrl;
+          requestOptions.data.configurationParam = configuration;
+        }
       }
-      else {
+      else if (homeSiteCount <= 1) {
         requestOptions.url += '/SetSPHSite';
+        requestOptions.data.sphSiteUrl = args.options.siteUrl;
```

Comment:
For the legacy API request, let's show a deprecation warning for the user telling them to use `spo homesite add` to add a new home site. In this future this command will only update existing home sites.